From 859a3ce92a1d8858f6789cea8aa136668e179280 Mon Sep 17 00:00:00 2001
From: Xaver Hugl <xaver.hugl@kde.org>
Date: Tue, 28 Apr 2026 13:41:30 +0200
Subject: [PATCH] libx264encoder: fix division by two

ceil-ing the result of an integer division doesn't do anything. Clearly this
was intended to use floating point math instead.
---
 src/libx264encoder.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/libx264encoder.cpp b/src/libx264encoder.cpp
index 8d959ab..1b8246c 100644
--- a/src/libx264encoder.cpp
+++ b/src/libx264encoder.cpp
@@ -61,8 +61,8 @@ bool LibX264Encoder::initialize(const QSize &size)
     // ensure we don't get errors, we need to ensure the size we set here is
     // even. We also insert a pad filter into the filter chain above to ensure
     // we don't end up padding with garbage.
-    m_avCodecContext->width = std::ceil(size.width() / 2) * 2;
-    m_avCodecContext->height = std::ceil(size.height() / 2) * 2;
+    m_avCodecContext->width = std::ceil(size.width() / 2.0) * 2;
+    m_avCodecContext->height = std::ceil(size.height() / 2.0) * 2;
     m_avCodecContext->max_b_frames = 0;
     m_avCodecContext->gop_size = 100;
     m_avCodecContext->pix_fmt = AV_PIX_FMT_YUV420P;
-- 
GitLab

